src/ProjectSystemTools/TableControl/TableToolWindow.cs

```diff
@@ -149,8 +155,11 @@ int IOleCommandTarget.Exec(ref Guid commandGroupGuid, uint commandId, uint comma
             return ((IOleCommandTarget)TableControl).Exec(ref commandGroupGuid, commandId, commandExecOption, pvaIn, pvaOut);
         }
 
-        protected override bool PreProcessMessage(ref Message m) =>
-            ContentWrapper.PreProcessMessage(ref m, this) || base.PreProcessMessage(ref m);
+        protected override bool PreProcessMessage(ref Message m)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
```

Comment:
`PreProcessMessage` is likely called a _lot_, and always on the UI thread. I think this guard should be removed and if an analyzer complains, it should be suppressed.